### ISSUE TITLE
e2e failure: SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance]

### ISSUE THREAD

**Issue by alex-mohr** [kubernetes/kubernetes#19895]
e2e failure: SchedulerPredicates [Serial] validates resource limits of pods that are allowed to run [Conformance]
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:355 Expected : 21 to equal : 22
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce/10440/

@saad-ali @hongchaodeng Recent issues/PRs mentioning the same test 
- #19487 
- #19495

**Comment by saad-ali** [comment]
Assigning to @hongchaodeng: Could you please take a look. Is this related at all to https://github.com/kubernetes/kubernetes/pull/19527?

Failed test log:
```
22:32:43 ------------------------------
22:32:43 SchedulerPredicates [Serial] 
22:32:43   validates resource limits of pods that are allowed to run [Conformance]
22:32:43   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:355
22:32:43 [BeforeEach] SchedulerPredicates [Serial]
22:32:43   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:69
22:32:43 STEP: Creating a kubernetes client
22:32:43 Jan 19 22:32:43.015: INFO: >>> testContext.KubeConfig: /var/lib/jenkins/jobs/kubernetes-e2e-gce/workspace/.kube/config
22:32:43 
22:32:43 STEP: Building a namespace api object
22:32:43 Jan 19 22:32:43.021: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-sched-pred-7g65r
22:32:43 Jan 19 22:32:43.023: INFO: Get service account default in ns e2e-tests-sched-pred-7g65r failed, ignoring for 2s: serviceaccounts "default" not found
22:32:45 Jan 19 22:32:45.026: INFO: Service account default in ns e2e-tests-sched-pred-7g65r with secrets found. (2.005391363s)
22:32:45 STEP: Waiting for a default service account to be provisioned in namespace
22:32:45 Jan 19 22:32:45.026: INFO: Waiting up to 2m0s for service account default to be provisioned in ns e2e-tests-sched-pred-7g65r
22:32:45 Jan 19 22:32:45.029: INFO: Service account default in ns e2e-tests-sched-pred-7g65r with secrets found. (2.410147ms)
22:32:45 [BeforeEach] SchedulerPredicates [Serial]
22:32:45   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:198
22:32:45 [It] validates resource limits of pods that are allowed to run [Conformance]
22:32:45   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:355
22:32:45 Jan 19 22:32:45.058: INFO: Pod elasticsearch-logging-v1-20ogt requesting capacity 100 on Node e2e-gce-minion-ltb2
22:32:45 Jan 19 22:32:45.058: INFO: Pod elasticsearch-logging-v1-yo99i requesting capacity 100 on Node e2e-gce-minion-jklx
22:32:45 Jan 19 22:32:45.058: INFO: Pod fluentd-elasticsearch-e2e-gce-minion-jklx requesting capacity 100 on Node e2e-gce-minion-jklx
22:32:45 Jan 19 22:32:45.058: INFO: Pod fluentd-elasticsearch-e2e-gce-minion-ltb2 requesting capacity 100 on Node e2e-gce-minion-ltb2
22:32:45 Jan 19 22:32:45.058: INFO: Pod fluentd-elasticsearch-e2e-gce-minion-nf16 requesting capacity 100 on Node e2e-gce-minion-nf16
22:32:45 Jan 19 22:32:45.058: INFO: Pod heapster-v11-bo7wq requesting capacity 100 on Node e2e-gce-minion-ltb2
22:32:45 Jan 19 22:32:45.058: INFO: Pod kibana-logging-v1-vsndq requesting capacity 100 on Node e2e-gce-minion-jklx
22:32:45 Jan 19 22:32:45.058: INFO: Pod kube-dns-v10-zk9r0 requesting capacity 310 on Node e2e-gce-minion-jklx
22:32:45 Jan 19 22:32:45.058: INFO: Pod kube-proxy-e2e-gce-minion-jklx requesting capacity 0 on Node e2e-gce-minion-jklx
22:32:45 Jan 19 22:32:45.058: INFO: Pod kube-proxy-e2e-gce-minion-ltb2 requesting capacity 0 on Node e2e-gce-minion-ltb2
22:32:45 Jan 19 22:32:45.058: INFO: Pod kube-proxy-e2e-gce-minion-nf16 requesting capacity 0 on Node e2e-gce-minion-nf16
22:32:45 Jan 19 22:32:45.058: INFO: Pod kube-ui-v4-0g8ve requesting capacity 100 on Node e2e-gce-minion-nf16
22:32:45 Jan 19 22:32:45.058: INFO: Pod l7-lb-controller-wasi9 requesting capacity 110 on Node e2e-gce-minion-nf16
22:32:45 Jan 19 22:32:45.058: INFO: Pod monitoring-influxdb-grafana-v2-o1b8e requesting capacity 200 on Node e2e-gce-minion-ltb2
22:32:45 Jan 19 22:32:45.058: INFO: Node: e2e-gce-minion-jklx has capacity: 1390
22:32:45 Jan 19 22:32:45.058: INFO: Node: e2e-gce-minion-ltb2 has capacity: 1500
22:32:45 Jan 19 22:32:45.058: INFO: Node: e2e-gce-minion-nf16 has capacity: 1690
22:32:45 STEP: Starting additional 8 Pods to fully saturate the cluster CPU and trying to start another one
22:32:46 Jan 19 22:32:46.068: INFO: 21 pods running
22:32:51 Jan 19 22:32:51.086: INFO: 21 pods running
22:32:56 Jan 19 22:32:56.104: INFO: 21 pods running
22:33:01 Jan 19 22:33:01.123: INFO: 21 pods running
22:33:06 Jan 19 22:33:06.141: INFO: 21 pods running
22:33:11 Jan 19 22:33:11.158: INFO: 21 pods running
22:33:16 Jan 19 22:33:16.201: INFO: 21 pods running
22:33:21 Jan 19 22:33:21.219: INFO: 21 pods running
22:33:26 Jan 19 22:33:26.236: INFO: 21 pods running
22:33:31 Jan 19 22:33:31.253: INFO: 21 pods running
22:33:36 Jan 19 22:33:36.271: INFO: 21 pods running
22:33:41 Jan 19 22:33:41.289: INFO: 21 pods running
22:33:46 Jan 19 22:33:46.306: INFO: 21 pods running
22:33:51 Jan 19 22:33:51.329: INFO: 21 pods running
22:33:56 Jan 19 22:33:56.346: INFO: 21 pods running
22:34:01 Jan 19 22:34:01.362: INFO: 21 pods running
22:34:06 Jan 19 22:34:06.379: INFO: 21 pods running
22:34:11 Jan 19 22:34:11.396: INFO: 21 pods running
22:34:16 Jan 19 22:34:16.412: INFO: 21 pods running
22:34:21 Jan 19 22:34:21.429: INFO: 21 pods running
22:34:26 Jan 19 22:34:26.445: INFO: 21 pods running
22:34:31 Jan 19 22:34:31.462: INFO: 21 pods running
22:34:36 Jan 19 22:34:36.479: INFO: 21 pods running
22:34:41 Jan 19 22:34:41.495: INFO: 21 pods running
22:34:46 Jan 19 22:34:46.534: INFO: 21 pods running
22:34:51 Jan 19 22:34:51.551: INFO: 21 pods running
22:34:56 Jan 19 22:34:56.573: INFO: 21 pods running
22:35:01 Jan 19 22:35:01.593: INFO: 21 pods running
22:35:06 Jan 19 22:35:06.614: INFO: 21 pods running
22:35:11 Jan 19 22:35:11.635: INFO: 21 pods running
22:35:16 Jan 19 22:35:16.652: INFO: 21 pods running
22:35:21 Jan 19 22:35:21.671: INFO: 21 pods running
22:35:26 Jan 19 22:35:26.688: INFO: 21 pods running
22:35:31 Jan 19 22:35:31.706: INFO: 21 pods running
22:35:36 Jan 19 22:35:36.723: INFO: 21 pods running
22:35:41 Jan 19 22:35:41.740: INFO: 21 pods running
22:35:46 Jan 19 22:35:46.759: INFO: 21 pods running
22:35:51 Jan 19 22:35:51.776: INFO: 21 pods running
22:35:56 Jan 19 22:35:56.793: INFO: 21 pods running
22:36:01 Jan 19 22:36:01.815: INFO: 21 pods running
22:36:06 Jan 19 22:36:06.835: INFO: 21 pods running
22:36:11 Jan 19 22:36:11.853: INFO: 21 pods running
22:36:16 Jan 19 22:36:16.870: INFO: 21 pods running
22:36:21 Jan 19 22:36:21.888: INFO: 21 pods running
22:36:26 Jan 19 22:36:26.904: INFO: 21 pods running
22:36:31 Jan 19 22:36:31.945: INFO: 21 pods running
22:36:36 Jan 19 22:36:36.961: INFO: 21 pods running
22:36:41 Jan 19 22:36:41.982: INFO: 21 pods running
22:36:47 Jan 19 22:36:47.000: INFO: 21 pods running
22:36:52 Jan 19 22:36:52.017: INFO: 21 pods running
22:36:57 Jan 19 22:36:57.034: INFO: 21 pods running
22:37:02 Jan 19 22:37:02.051: INFO: 21 pods running
22:37:07 Jan 19 22:37:07.073: INFO: 21 pods running
22:37:12 Jan 19 22:37:12.091: INFO: 21 pods running
22:37:17 Jan 19 22:37:17.109: INFO: 21 pods running
22:37:22 Jan 19 22:37:22.126: INFO: 21 pods running
22:37:27 Jan 19 22:37:27.143: INFO: 21 pods running
22:37:32 Jan 19 22:37:32.160: INFO: 21 pods running
22:37:37 Jan 19 22:37:37.177: INFO: 21 pods running
22:37:42 Jan 19 22:37:42.221: INFO: 21 pods running
22:37:47 Jan 19 22:37:47.238: INFO: 21 pods running
22:37:52 Jan 19 22:37:52.256: INFO: 21 pods running
22:37:57 Jan 19 22:37:57.272: INFO: 21 pods running
22:38:02 Jan 19 22:38:02.290: INFO: 21 pods running
22:38:07 Jan 19 22:38:07.307: INFO: 21 pods running
22:38:12 Jan 19 22:38:12.329: INFO: 21 pods running
22:38:17 Jan 19 22:38:17.346: INFO: 21 pods running
22:38:22 Jan 19 22:38:22.364: INFO: 21 pods running
22:38:27 Jan 19 22:38:27.383: INFO: 21 pods running
22:38:32 Jan 19 22:38:32.400: INFO: 21 pods running
22:38:37 Jan 19 22:38:37.418: INFO: 21 pods running
22:38:42 Jan 19 22:38:42.438: INFO: 21 pods running
22:38:47 Jan 19 22:38:47.455: INFO: 21 pods running
22:38:52 Jan 19 22:38:52.472: INFO: 21 pods running
22:38:57 Jan 19 22:38:57.491: INFO: 21 pods running
22:39:02 Jan 19 22:39:02.509: INFO: 21 pods running
22:39:07 Jan 19 22:39:07.554: INFO: 21 pods running
22:39:12 Jan 19 22:39:12.571: INFO: 21 pods running
22:39:17 Jan 19 22:39:17.593: INFO: 21 pods running
22:39:22 Jan 19 22:39:22.611: INFO: 21 pods running
22:39:27 Jan 19 22:39:27.628: INFO: 21 pods running
22:39:32 Jan 19 22:39:32.645: INFO: 21 pods running
22:39:37 Jan 19 22:39:37.663: INFO: 21 pods running
22:39:42 Jan 19 22:39:42.680: INFO: 21 pods running
22:39:47 Jan 19 22:39:47.696: INFO: 21 pods running
22:39:52 Jan 19 22:39:52.713: INFO: 21 pods running
22:39:57 Jan 19 22:39:57.748: INFO: 21 pods running
22:40:02 Jan 19 22:40:02.766: INFO: 21 pods running
22:40:07 Jan 19 22:40:07.783: INFO: 21 pods running
22:40:12 Jan 19 22:40:12.800: INFO: 21 pods running
22:40:17 Jan 19 22:40:17.817: INFO: 21 pods running
22:40:22 Jan 19 22:40:22.840: INFO: 21 pods running
22:40:27 Jan 19 22:40:27.858: INFO: 21 pods running
22:40:32 Jan 19 22:40:32.874: INFO: 21 pods running
22:40:37 Jan 19 22:40:37.891: INFO: 21 pods running
22:40:42 Jan 19 22:40:42.908: INFO: 21 pods running
22:40:47 Jan 19 22:40:47.923: INFO: 21 pods running
22:40:52 Jan 19 22:40:52.940: INFO: 21 pods running
22:40:57 Jan 19 22:40:57.957: INFO: 21 pods running
22:41:02 Jan 19 22:41:02.974: INFO: 21 pods running
22:41:07 Jan 19 22:41:07.991: INFO: 21 pods running
22:41:13 Jan 19 22:41:13.029: INFO: 21 pods running
22:41:18 Jan 19 22:41:18.045: INFO: 21 pods running
22:41:23 Jan 19 22:41:23.062: INFO: 21 pods running
22:41:28 Jan 19 22:41:28.085: INFO: 21 pods running
22:41:33 Jan 19 22:41:33.102: INFO: 21 pods running
22:41:38 Jan 19 22:41:38.119: INFO: 21 pods running
22:41:43 Jan 19 22:41:43.134: INFO: 21 pods running
22:41:48 Jan 19 22:41:48.152: INFO: 21 pods running
22:41:53 Jan 19 22:41:53.168: INFO: 21 pods running
22:41:58 Jan 19 22:41:58.194: INFO: 21 pods running
22:42:03 Jan 19 22:42:03.211: INFO: 21 pods running
22:42:08 Jan 19 22:42:08.229: INFO: 21 pods running
22:42:13 Jan 19 22:42:13.313: INFO: 21 pods running
22:42:18 Jan 19 22:42:18.334: INFO: 21 pods running
22:42:23 Jan 19 22:42:23.364: INFO: 21 pods running
22:42:28 Jan 19 22:42:28.382: INFO: 21 pods running
22:42:33 Jan 19 22:42:33.404: INFO: 21 pods running
22:42:38 Jan 19 22:42:38.421: INFO: 21 pods running
22:42:43 Jan 19 22:42:43.437: INFO: 21 pods running
22:42:48 Jan 19 22:42:48.454: INFO: 21 pods running
22:42:48 Jan 19 22:42:48.454: INFO: Timed out after 10m0s waiting for pods to start running.
22:42:48 [AfterEach] SchedulerPredicates [Serial]
22:42:48   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:190
22:42:48 [AfterEach] SchedulerPredicates [Serial]
22:42:48   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:70
22:42:48 STEP: Collecting events from namespace "e2e-tests-sched-pred-7g65r".
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-0: {default-scheduler } Scheduled: Successfully assigned overcommit-0 to e2e-gce-minion-jklx
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-0: {kubelet e2e-gce-minion-jklx} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-0: {kubelet e2e-gce-minion-jklx} Created: Created container with docker id 868a0c936547
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-0: {kubelet e2e-gce-minion-jklx} Started: Started container with docker id 868a0c936547
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-1: {default-scheduler } Scheduled: Successfully assigned overcommit-1 to e2e-gce-minion-jklx
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-1: {kubelet e2e-gce-minion-jklx} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-1: {kubelet e2e-gce-minion-jklx} Created: Created container with docker id 306e592047f3
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-1: {kubelet e2e-gce-minion-jklx} Started: Started container with docker id 306e592047f3
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-2: {default-scheduler } Scheduled: Successfully assigned overcommit-2 to e2e-gce-minion-ltb2
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-2: {kubelet e2e-gce-minion-ltb2} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-2: {kubelet e2e-gce-minion-ltb2} Created: Created container with docker id dace8fe3deac
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-2: {kubelet e2e-gce-minion-ltb2} Started: Started container with docker id dace8fe3deac
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-3: {default-scheduler } Scheduled: Successfully assigned overcommit-3 to e2e-gce-minion-ltb2
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-3: {kubelet e2e-gce-minion-ltb2} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-3: {kubelet e2e-gce-minion-ltb2} Created: Created container with docker id 388265f60a11
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-3: {kubelet e2e-gce-minion-ltb2} Started: Started container with docker id 388265f60a11
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-4: {default-scheduler } Scheduled: Successfully assigned overcommit-4 to e2e-gce-minion-nf16
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-4: {kubelet e2e-gce-minion-nf16} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-4: {kubelet e2e-gce-minion-nf16} Created: Created container with docker id 04adfce4bf89
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-4: {kubelet e2e-gce-minion-nf16} Started: Started container with docker id 04adfce4bf89
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-5: {default-scheduler } Scheduled: Successfully assigned overcommit-5 to e2e-gce-minion-nf16
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-5: {kubelet e2e-gce-minion-nf16} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-5: {kubelet e2e-gce-minion-nf16} Created: Created container with docker id fa58962a7e41
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-5: {kubelet e2e-gce-minion-nf16} Started: Started container with docker id fa58962a7e41
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-6: {default-scheduler } Scheduled: Successfully assigned overcommit-6 to e2e-gce-minion-nf16
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-6: {kubelet e2e-gce-minion-nf16} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-6: {kubelet e2e-gce-minion-nf16} Created: Created container with docker id 39ce1a9811f4
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-6: {kubelet e2e-gce-minion-nf16} Started: Started container with docker id 39ce1a9811f4
22:42:48 Jan 19 22:42:48.541: INFO: event for overcommit-7: {default-scheduler } FailedScheduling: Failed for reason CPU and possibly others
22:42:48 Jan 19 22:42:48.558: INFO: POD                                        NODE                 PHASE    GRACE  CONDITIONS
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-0                               e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-1                               e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-2                               e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-3                               e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-4                               e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-5                               e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-6                               e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: overcommit-7                                                    Pending         []
22:42:48 Jan 19 22:42:48.558: INFO: elasticsearch-logging-v1-20ogt             e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:38:51 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: elasticsearch-logging-v1-yo99i             e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:54:49 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: fluentd-elasticsearch-e2e-gce-minion-jklx  e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:54:49 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: fluentd-elasticsearch-e2e-gce-minion-ltb2  e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:37:54 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: fluentd-elasticsearch-e2e-gce-minion-nf16  e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:37:55 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: heapster-v11-bo7wq                         e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:38:29 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: kibana-logging-v1-vsndq                    e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:05 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: kube-dns-v10-zk9r0                         e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:55:00 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: kube-proxy-e2e-gce-minion-jklx             e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:54:45 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: kube-proxy-e2e-gce-minion-ltb2             e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:37:30 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: kube-proxy-e2e-gce-minion-nf16             e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:37:31 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: kube-ui-v4-0g8ve                           e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:38:21 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: l7-lb-controller-wasi9                     e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:38:33 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: monitoring-influxdb-grafana-v2-o1b8e       e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 21:39:04 -0800 PST  }]
22:42:48 Jan 19 22:42:48.558: INFO: 
22:42:48 Jan 19 22:42:48.563: INFO: 
22:42:48 Logging node info for node e2e-gce-minion-jklx
22:42:48 Jan 19 22:42:48.567: INFO: Node Info: &{{ } {e2e-gce-minion-jklx   /api/v1/nodes/e2e-gce-minion-jklx cc61ded7-bf37-11e5-8634-42010af00002 8888 0 2016-01-19 21:36:47 -0800 PST <nil> <nil> map[failure-domain.alpha.kubernetes.io/zone:us-central1-f kubernetes.io/e2e-0879f39b-bf39-11e5-9681-42010af01555:42 kubernetes.io/hostname:e2e-gce-minion-jklx failure-domain.alpha.kubernetes.io/region:us-central1] map[]} {10.245.72.0/24 10824444563001539551 gce://k8s-jkns-e2e-gce/us-central1-f/e2e-gce-minion-jklx false} {map[cpu:{2.000 DecimalSI} memory:{7863918592.000 BinarySI} pods:{110.000 DecimalSI}] map[cpu:{2.000 DecimalSI} memory:{7863918592.000 BinarySI} pods:{110.000 DecimalSI}]  [{OutOfDisk False 2016-01-19 22:42:43 -0800 PST 2016-01-19 22:16:33 -0800 PST KubeletHasSufficientDisk kubelet has sufficient disk space available} {Ready True 2016-01-19 22:42:43 -0800 PST 2016-01-19 22:30:44 -0800 PST KubeletReady kubelet is posting ready status}] [{InternalIP 10.240.0.3} {ExternalIP 104.197.222.15}] {{10250}} { DAF73D06-8DD2-691E-A48F-1D86CE92D391 8251fed1-1553-4395-abd7-90c32d080f82 3.16.0-0.bpo.4-amd64 Debian GNU/Linux 7 (wheezy) docker://1.8.3 v1.2.0-alpha.6.230+ecf60387206d93 v1.2.0-alpha.6.230+ecf60387206d93} [{[gcr.io/google_containers/kube-proxy:2dfa6411b2719f3fb317fd60d62fdc75] 162767061} {[busybox:latest] 1113554} {[redis:latest] 151314673} {[nginx:latest] 133864957} {[gcr.io/google_containers/kube2sky:1.12] 24482187} {[gcr.io/google_containers/elasticsearch:1.8] 410989305} {[gcr.io/google_containers/mounttest:0.6] 2084693} {[gcr.io/google_containers/hostexec:1.2] 13209617} {[gcr.io/google_containers/mounttest-user:0.3] 1718853} {[gcr.io/google_containers/mounttest:0.5] 1718853} {[gcr.io/google_containers/skydns:2015-10-13-8c72f8c] 40551394} {[gcr.io/google_containers/fluentd-elasticsearch:1.12] 551579914} {[gcr.io/google_containers/pause:2.0] 350164} {[gcr.io/google_containers/porter:cd5cb5791ebaa8641955f0e8c2a9bed669b1eaab] 5010921} {[gcr.io/google_containers/netexec:1.3.1] 6530936} {[gcr.io/google_containers/goproxy:0.1] 5489816} {[gcr.io/google_containers/portforwardtester:1.0] 2296329} {[gcr.io/google_samples/gb-frontend:v3] 509707703} {[gcr.io/google_samples/gb-redisslave:v1] 109508753} {[gcr.io/google_containers/nettest:1.6] 10374838} {[gcr.io/google_containers/exechealthz:1.0] 7099444} {[gcr.io/google_containers/jessie-dnsutils:latest] 190148402} {[gcr.io/google_containers/mounttest:0.2] 1752375} {[gcr.io/google_containers/kibana:1.3] 396897764} {[gcr.io/google_containers/dnsutils:latest] 141895666} {[gcr.io/google_containers/etcd:2.0.9] 12819040} {[gcr.io/google_containers/pause:0.8.0] 241656} {[gcr.io/google_containers/eptest:0.1] 2970692} {[b.gcr.io/k8s_authenticated_test/serve_hostname:1.1 gcr.io/google_containers/serve_hostname:1.1] 4522409} {[gcr.io/google_containers/update-demo:kitten] 4549069} {[gcr.io/google_containers/update-demo:nautilus] 4555533} {[gcr.io/google_containers/nginx:1.7.9] 91664166} {[gcr.io/google_containers/test-webserver:latest] 4534272} {[gcr.io/google_containers/busybox:latest] 2433303} {[gcr.io/google_containers/liveness:latest] 4387474}]}}
22:42:48 Jan 19 22:42:48.623: INFO: 
22:42:48 Logging kubelet events for node e2e-gce-minion-jklx
22:42:48 Jan 19 22:42:48.631: INFO: 
22:42:48 Logging pods the kubelet thinks is on node e2e-gce-minion-jklx
22:42:48 Jan 19 22:42:48.735: INFO: overcommit-1 started at <nil> (0 container statuses recorded)
22:42:48 Jan 19 22:42:48.735: INFO: overcommit-0 started at <nil> (0 container statuses recorded)
22:42:48 Jan 19 22:42:48.735: INFO: kibana-logging-v1-vsndq started at <nil> (0 container statuses recorded)
22:42:48 Jan 19 22:42:48.735: INFO: kube-dns-v10-zk9r0 started at <nil> (0 container statuses recorded)
22:42:48 Jan 19 22:42:48.735: INFO: fluentd-elasticsearch-e2e-gce-minion-jklx started at <nil> (0 container statuses recorded)
22:42:48 Jan 19 22:42:48.735: INFO: elasticsearch-logging-v1-yo99i started at <nil> (0 container statuses recorded)
22:42:48 Jan 19 22:42:48.735: INFO: kube-proxy-e2e-gce-minion-jklx started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:48.972: INFO: ERROR kubelet_docker_errors{operation_type="info"} => 1 @[0]
22:42:49 Jan 19 22:42:49.056: INFO: ERROR kubelet_docker_errors{operation_type="inspect_image"} => 3 @[0]
22:42:49 Jan 19 22:42:49.056: INFO: ERROR kubelet_docker_errors{operation_type="list_containers"} => 11 @[0]
22:42:49 Jan 19 22:42:49.056: INFO: ERROR kubelet_docker_errors{operation_type="list_images"} => 5 @[0]
22:42:49 Jan 19 22:42:49.056: INFO: ERROR kubelet_docker_errors{operation_type="stop_container"} => 11 @[0]
22:42:49 Jan 19 22:42:49.056: INFO: ERROR kubelet_docker_errors{operation_type="version"} => 16 @[0]
22:42:49 Jan 19 22:42:49.056: INFO: 
22:42:49 Latency metrics for node e2e-gce-minion-jklx
22:42:49 Jan 19 22:42:49.056: INFO: 
22:42:49 Logging node info for node e2e-gce-minion-ltb2
22:42:49 Jan 19 22:42:49.060: INFO: Node Info: &{{ } {e2e-gce-minion-ltb2   /api/v1/nodes/e2e-gce-minion-ltb2 cd3d1635-bf37-11e5-8634-42010af00002 8886 0 2016-01-19 21:36:49 -0800 PST <nil> <nil> map[failure-domain.alpha.kubernetes.io/zone:us-central1-f kubernetes.io/hostname:e2e-gce-minion-ltb2 failure-domain.alpha.kubernetes.io/region:us-central1] map[]} {10.245.34.0/24 1650113100415390874 gce://k8s-jkns-e2e-gce/us-central1-f/e2e-gce-minion-ltb2 false} {map[memory:{7863918592.000 BinarySI} pods:{110.000 DecimalSI} cpu:{2.000 DecimalSI}] map[pods:{110.000 DecimalSI} cpu:{2.000 DecimalSI} memory:{7863918592.000 BinarySI}]  [{OutOfDisk False 2016-01-19 22:42:39 -0800 PST 2016-01-19 22:16:34 -0800 PST KubeletHasSufficientDisk kubelet has sufficient disk space available} {Ready True 2016-01-19 22:42:39 -0800 PST 2016-01-19 22:30:43 -0800 PST KubeletReady kubelet is posting ready status}] [{InternalIP 10.240.0.5} {ExternalIP 104.197.212.161}] {{10250}} { 178CDFF8-7486-2695-B9B8-4123FF07C93C 8686c58a-0898-4596-ac1c-73c018cc081e 3.16.0-0.bpo.4-amd64 Debian GNU/Linux 7 (wheezy) docker://1.8.3 v1.2.0-alpha.6.230+ecf60387206d93 v1.2.0-alpha.6.230+ecf60387206d93} [{[gcr.io/google_containers/kube-proxy:2dfa6411b2719f3fb317fd60d62fdc75] 162767061} {[gcr.io/google_containers/heapster:v0.18.4] 37026444} {[gcr.io/google_containers/elasticsearch:1.8] 410989305} {[gcr.io/google_containers/heapster_grafana:v2.1.1] 206641132} {[gcr.io/google_containers/fluentd-elasticsearch:1.12] 551579914} {[gcr.io/google_containers/pause:2.0] 350164} {[gcr.io/google_containers/heapster_influxdb:v0.4] 271818313} {[gcr.io/google_containers/netexec:1.0] 7280393} {[gcr.io/google_samples/gb-frontend:v3] 509707703} {[gcr.io/google_samples/gb-redisslave:v1] 109508753} {[gcr.io/google_containers/pause:0.8.0] 241656} {[gcr.io/google_containers/serve_hostname:1.1] 4522409} {[gcr.io/google_containers/update-demo:nautilus] 4555533} {[gcr.io/google_containers/busybox:latest] 2433303}]}}
22:42:49 Jan 19 22:42:49.060: INFO: 
22:42:49 Logging kubelet events for node e2e-gce-minion-ltb2
22:42:49 Jan 19 22:42:49.067: INFO: 
22:42:49 Logging pods the kubelet thinks is on node e2e-gce-minion-ltb2
22:42:49 Jan 19 22:42:49.091: INFO: fluentd-elasticsearch-e2e-gce-minion-ltb2 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.091: INFO: elasticsearch-logging-v1-20ogt started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.091: INFO: kube-proxy-e2e-gce-minion-ltb2 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.091: INFO: overcommit-3 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.091: INFO: overcommit-2 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.091: INFO: monitoring-influxdb-grafana-v2-o1b8e started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.091: INFO: heapster-v11-bo7wq started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.156: INFO: ERROR kubelet_docker_errors{operation_type="info"} => 1 @[0]
22:42:49 Jan 19 22:42:49.156: INFO: ERROR kubelet_docker_errors{operation_type="list_containers"} => 12 @[0]
22:42:49 Jan 19 22:42:49.156: INFO: ERROR kubelet_docker_errors{operation_type="list_images"} => 5 @[0]
22:42:49 Jan 19 22:42:49.156: INFO: ERROR kubelet_docker_errors{operation_type="version"} => 18 @[0]
22:42:49 Jan 19 22:42:49.156: INFO: 
22:42:49 Latency metrics for node e2e-gce-minion-ltb2
22:42:49 Jan 19 22:42:49.156: INFO: 
22:42:49 Logging node info for node e2e-gce-minion-nf16
22:42:49 Jan 19 22:42:49.221: INFO: Node Info: &{{ } {e2e-gce-minion-nf16   /api/v1/nodes/e2e-gce-minion-nf16 cc62bda1-bf37-11e5-8634-42010af00002 8887 0 2016-01-19 21:36:47 -0800 PST <nil> <nil> map[failure-domain.alpha.kubernetes.io/region:us-central1 failure-domain.alpha.kubernetes.io/zone:us-central1-f kubernetes.io/hostname:e2e-gce-minion-nf16] map[]} {10.255.28.0/24 12637447196558872400 gce://k8s-jkns-e2e-gce/us-central1-f/e2e-gce-minion-nf16 false} {map[cpu:{2.000 DecimalSI} memory:{7863918592.000 BinarySI} pods:{110.000 DecimalSI}] map[memory:{7863918592.000 BinarySI} pods:{110.000 DecimalSI} cpu:{2.000 DecimalSI}]  [{OutOfDisk False 2016-01-19 22:42:43 -0800 PST 2016-01-19 22:16:34 -0800 PST KubeletHasSufficientDisk kubelet has sufficient disk space available} {Ready True 2016-01-19 22:42:43 -0800 PST 2016-01-19 22:30:39 -0800 PST KubeletReady kubelet is posting ready status}] [{InternalIP 10.240.0.4} {ExternalIP 104.197.206.180}] {{10250}} { 8BB2840B-1CA3-D50B-0F3B-6E7150784932 6bbef892-07db-4c6d-a1a5-5193fb340af0 3.16.0-0.bpo.4-amd64 Debian GNU/Linux 7 (wheezy) docker://1.8.3 v1.2.0-alpha.6.230+ecf60387206d93 v1.2.0-alpha.6.230+ecf60387206d93} [{[gcr.io/google_containers/kube-proxy:2dfa6411b2719f3fb317fd60d62fdc75] 162767061} {[redis:latest] 151314673} {[nginx:latest] 133864957} {[gcr.io/google_containers/kube-ui:v4] 5676927} {[gcr.io/google_containers/glbc:0.5.1] 224751780} {[gcr.io/google_containers/mounttest:0.6] 2084693} {[gcr.io/google_containers/mounttest-user:0.3] 1718853} {[gcr.io/google_containers/defaultbackend:1.0] 7513643} {[gcr.io/google_containers/mounttest:0.5] 1718853} {[gcr.io/google_containers/fluentd-elasticsearch:1.12] 551579914} {[gcr.io/google_containers/pause:2.0] 350164} {[gcr.io/google_containers/netexec:1.0] 7280393} {[gcr.io/google_containers/pause:0.8.0] 241656} {[gcr.io/google_containers/serve_hostname:1.1] 4522409} {[gcr.io/google_containers/busybox:latest] 2433303}]}}
22:42:49 Jan 19 22:42:49.221: INFO: 
22:42:49 Logging kubelet events for node e2e-gce-minion-nf16
22:42:49 Jan 19 22:42:49.424: INFO: 
22:42:49 Logging pods the kubelet thinks is on node e2e-gce-minion-nf16
22:42:49 Jan 19 22:42:49.656: INFO: kube-ui-v4-0g8ve started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.656: INFO: kube-proxy-e2e-gce-minion-nf16 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.656: INFO: overcommit-6 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.656: INFO: overcommit-5 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.656: INFO: overcommit-4 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.656: INFO: l7-lb-controller-wasi9 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.656: INFO: fluentd-elasticsearch-e2e-gce-minion-nf16 started at <nil> (0 container statuses recorded)
22:42:49 Jan 19 22:42:49.910: INFO: ERROR kubelet_docker_errors{operation_type="info"} => 1 @[0]
22:42:49 Jan 19 22:42:49.910: INFO: ERROR kubelet_docker_errors{operation_type="list_containers"} => 12 @[0]
22:42:49 Jan 19 22:42:49.910: INFO: ERROR kubelet_docker_errors{operation_type="list_images"} => 5 @[0]
22:42:49 Jan 19 22:42:49.910: INFO: ERROR kubelet_docker_errors{operation_type="version"} => 18 @[0]
22:42:49 Jan 19 22:42:49.911: INFO: 
22:42:49 Latency metrics for node e2e-gce-minion-nf16
22:42:49 Jan 19 22:42:49.911: INFO: Waiting up to 1m0s for all nodes to be ready
22:42:50 STEP: Destroying namespace "e2e-tests-sched-pred-7g65r" for this suite.
22:42:50 
22:42:50 • Failure [607.605 seconds]
22:42:50 SchedulerPredicates [Serial]
22:42:50 /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:473
22:42:50   validates resource limits of pods that are allowed to run [Conformance] [It]
22:42:50   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:355
22:42:50 
22:42:50   Expected
22:42:50       <int>: 21
22:42:50   to equal
22:42:50       <int>: 22
22:42:50 
22:42:50   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:86
22:42:50 ------------------------------

```

**Comment by hongchaodeng** [comment]
Here is my analysis:

Looking at the log, it showed the error:
```
FailedScheduling: Failed for reason CPU and possibly others
```
The error is returned [here](https://github.com/kubernetes/kubernetes/blob/252a956150e7862255961007c74a31cd89a362bf/plugin/pkg/scheduler/generic_scheduler.go#L91). It means the pod can't be assigned to any node and some of nodes didn't have enough CPU.

Last time, the problem came from [here](https://github.com/kubernetes/kubernetes/blob/252a956150e7862255961007c74a31cd89a362bf/plugin/pkg/scheduler/generic_scheduler.go#L86). If a node doesn't have enough CPU, it returns error immediately without search the rest of nodes. It seems the problem is fixed based on the return error.

Re. the test, it tries to validate that no more pods to run beyond CPU limit, so the reason of CPU is also expected. But somehow it shouldn't fail all nodes. Limited by the information of the log, I didn't know why it failed all nodes.

I couldn't do any analysis further based on existing information. I would need more logging [here](https://github.com/kubernetes/kubernetes/blob/252a956150e7862255961007c74a31cd89a362bf/plugin/pkg/scheduler/generic_scheduler.go#L130) to print out why each FitPredicate failed on each node to understand the failure in depth.

**Comment by hongchaodeng** [comment]
Again, I can't reproduce it in running that single test.
We at CoreOS have our jenkins jobs running on GCE. We should be able to run and get more logs if the flaky error happened.
At the mean time, it would be great if we can put more logging into upstream. That would help us debug and understand easier the behavior of scheduler finding fitting nodes. What do you think? @alex-mohr 
/cc @yifan-gu

**Comment by yifan-gu** [comment]
@hongchaodeng @alex-mohr What's the job name of this e2e test? kubernetes-e2e-gce?

**Comment by hongchaodeng** [comment]
Create a PR #19906 to return details of failed predicate map.
IMHO that would be very helpful.

**Comment by hongchaodeng** [comment]
One more thing suspicious:
There is a node has exactly 1500 cpu for 3 pods of 500 cpu
```
Node: e2e-gce-minion-ltb2 has capacity: 1500
```

The log shows that such edge case only schedules 2 pods:
```
overcommit-0                               e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
overcommit-1                               e2e-gce-minion-jklx  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
overcommit-2                               e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
overcommit-3                               e2e-gce-minion-ltb2  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
overcommit-4                               e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
overcommit-5                               e2e-gce-minion-nf16  Running         [{Ready True 0001-01-01 00:00:00 +0000 UTC 2016-01-19 22:32:46 -0800 PST  }]
overcommit-6                               e2e-gce-minion-nf16  R
overcommit-7                                                    Pending         []
```

**Comment by hongchaodeng** [comment]
Create #19910 print out details of resource request/used/capacity information in case of failure

**Comment by hongchaodeng** [comment]
In summary of above messages:
- the problem doesn't seem to be directly caused by #19527 
- I need both #19906 and #19910 to get in and when such failure happened again I should be able to analyze it further.

**Comment by saad-ali** [comment]
@hongchaodeng Thanks for digging in! I'll review and approve those PRs adding additional logging needed to debug this.

**Comment by thockin** [comment]
Can we get a status update on this issue?

**Comment by hongchaodeng** [comment]
Hi @thockin ,

Not sure if you asking me. I would answer anyway.

Last time I debugged, I couldn't get enough information to find out root cause. I'm blocked by two conditions:

- I need both #19906 and #19910 to have more information. I think #19910 is failed by flaky tests.
- After those two PRs, I need the logs of the failure to do further analysis. Unfortunately I didn't get such failure in our test infrastructure limited by the number of runs and probably inconsistent setup. I was expecting Google folks could provide it if the failure took place again.

**Comment by thockin** [comment]
Both PRs are in.

**Comment by davidopp** [comment]
Most recent failure is
http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-serial/351/testReport/
which was after #19906 was merged but not after #19910 was merged.

**Comment by gmarek** [comment]
I looked at it for few minutes and it's not Scheduler's or tests bug. It's a problem with Scheduler-Kubelet synchronization. Note logs:
```
07:37:20 Jan 31 07:37:20.857: INFO: Pod elasticsearch-logging-v1-55d0f requesting capacity 100 on Node e2e-serial-minion-plex
07:37:20 Jan 31 07:37:20.857: INFO: Pod elasticsearch-logging-v1-qhuoe requesting capacity 100 on Node e2e-serial-minion-z20s
07:37:20 Jan 31 07:37:20.857: INFO: Pod fluentd-elasticsearch-e2e-serial-minion-plex requesting capacity 100 on Node e2e-serial-minion-plex
07:37:20 Jan 31 07:37:20.857: INFO: Pod fluentd-elasticsearch-e2e-serial-minion-z20s requesting capacity 100 on Node e2e-serial-minion-z20s
07:37:20 Jan 31 07:37:20.857: INFO: Pod heapster-v12-uxbpp requesting capacity 100 on Node e2e-serial-minion-plex
07:37:20 Jan 31 07:37:20.857: INFO: Pod kibana-logging-v1-t28j4 requesting capacity 100 on Node e2e-serial-minion-plex
07:37:20 Jan 31 07:37:20.857: INFO: Pod kube-dns-v10-l2dvi requesting capacity 310 on Node e2e-serial-minion-z20s
07:37:20 Jan 31 07:37:20.857: INFO: Pod kube-proxy-e2e-serial-minion-plex requesting capacity 0 on Node e2e-serial-minion-plex
07:37:20 Jan 31 07:37:20.857: INFO: Pod kube-proxy-e2e-serial-minion-z20s requesting capacity 0 on Node e2e-serial-minion-z20s
07:37:20 Jan 31 07:37:20.857: INFO: Pod kube-ui-v5-50or4 requesting capacity 100 on Node e2e-serial-minion-z20s
07:37:20 Jan 31 07:37:20.857: INFO: Pod l7-lb-controller-530qx requesting capacity 110 on Node e2e-serial-minion-z20s
07:37:20 Jan 31 07:37:20.857: INFO: Pod monitoring-influxdb-grafana-v2-0hdan requesting capacity 200 on Node e2e-serial-minion-plex
07:37:20 Jan 31 07:37:20.857: INFO: Node: e2e-serial-minion-mue1 has capacity: 2000
07:37:20 Jan 31 07:37:20.857: INFO: Node: e2e-serial-minion-plex has capacity: 1400
07:37:20 Jan 31 07:37:20.857: INFO: Node: e2e-serial-minion-z20s has capacity: 1280
07:37:20 STEP: Starting additional 8 Pods to fully saturate the cluster CPU and trying to start another one
07:37:21 Jan 31 07:37:21.807: INFO: 20 pods running
07:37:26 Jan 31 07:37:26.812: INFO: Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.
07:37:37 [AfterEach] SchedulerPredicates [Serial]
07:37:37   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/scheduler_predicates.go:190
07:37:37 [AfterEach] SchedulerPredicates [Serial]
07:37:37   /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework.go:70
07:37:37 STEP: Collecting events from namespace "e2e-tests-sched-pred-3atsz".
07:37:37 Jan 31 07:37:37.132: INFO: event for additional-pod: {default-scheduler } Scheduled: Successfully assigned additional-pod to e2e-serial-minion-mue1
07:37:37 Jan 31 07:37:37.132: INFO: event for additional-pod: {kubelet e2e-serial-minion-mue1} InsufficientFreeCPU: cannot start the pod due to insufficient free CPU.
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-0: {default-scheduler } Scheduled: Successfully assigned overcommit-0 to e2e-serial-minion-mue1
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-0: {kubelet e2e-serial-minion-mue1} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-0: {kubelet e2e-serial-minion-mue1} Created: Created container with docker id e53c1587af51
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-0: {kubelet e2e-serial-minion-mue1} Started: Started container with docker id e53c1587af51
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-1: {default-scheduler } Scheduled: Successfully assigned overcommit-1 to e2e-serial-minion-mue1
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-1: {kubelet e2e-serial-minion-mue1} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-1: {kubelet e2e-serial-minion-mue1} Created: Created container with docker id 69a8c59f541a
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-1: {kubelet e2e-serial-minion-mue1} Started: Started container with docker id 69a8c59f541a
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-2: {default-scheduler } Scheduled: Successfully assigned overcommit-2 to e2e-serial-minion-plex
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-2: {kubelet e2e-serial-minion-plex} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-2: {kubelet e2e-serial-minion-plex} Created: Created container with docker id dcb05462f112
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-2: {kubelet e2e-serial-minion-plex} Started: Started container with docker id dcb05462f112
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-3: {default-scheduler } Scheduled: Successfully assigned overcommit-3 to e2e-serial-minion-z20s
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-3: {kubelet e2e-serial-minion-z20s} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
07:37:37 Jan 31 07:37:37.132: INFO: event for overcommit-3: {kubelet e2e-serial-minion-z20s} Created: Created container with docker id 0f754b1dba8b
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-3: {kubelet e2e-serial-minion-z20s} Started: Started container with docker id 0f754b1dba8b
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-4: {default-scheduler } Scheduled: Successfully assigned overcommit-4 to e2e-serial-minion-mue1
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-4: {kubelet e2e-serial-minion-mue1} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-4: {kubelet e2e-serial-minion-mue1} Created: Created container with docker id 65176e3f5e91
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-4: {kubelet e2e-serial-minion-mue1} Started: Started container with docker id 65176e3f5e91
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-5: {default-scheduler } Scheduled: Successfully assigned overcommit-5 to e2e-serial-minion-plex
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-5: {kubelet e2e-serial-minion-plex} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-5: {kubelet e2e-serial-minion-plex} Created: Created container with docker id dbe82c72cbcb
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-5: {kubelet e2e-serial-minion-plex} Started: Started container with docker id dbe82c72cbcb
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-6: {default-scheduler } Scheduled: Successfully assigned overcommit-6 to e2e-serial-minion-z20s
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-6: {kubelet e2e-serial-minion-z20s} Pulled: Container image "gcr.io/google_containers/pause:2.0" already present on machine
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-6: {kubelet e2e-serial-minion-z20s} Created: Created container with docker id b1fdc63bb4b1
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-6: {kubelet e2e-serial-minion-z20s} Started: Started container with docker id b1fdc63bb4b1
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-7: {default-scheduler } Scheduled: Successfully assigned overcommit-7 to e2e-serial-minion-mue1
07:37:37 Jan 31 07:37:37.133: INFO: event for overcommit-7: {kubelet e2e-serial-minion-mue1} InsufficientFreeCPU: cannot start the pod due to insufficient free CPU.
```

No system pods are assigned to mue1 Node at the beginning of the test leaving it with full 2000 millicores, which can fit exactly 4 overcommit pods (each is using 500m). Scheduler actually behaves correctly and in one of last lines assigns overcommit-7 pod to mue1 Node, but it's not started there, as there's not enough free resources. My guess is that some mirror pod was created afterwards screwing with Scheduler's knowledge.

I'm not quite sure how to proceed. Easy ducttape is to make sure all system pods are running after reboot tests, but I do believe that this deserves some proper handling, as it seems that we have a race between scheduler and kubelet.

cc @kubernetes/goog-node

**Comment by dchen1107** [comment]
Haven't dig into the detail yet. But with respect to @gmarek's comment at  https://github.com/kubernetes/kubernetes/issues/19895#issuecomment-178462848, shouldn't we start test after the cluster is stabled? This means the only potential mirror pod for test environment -- fluentd is running on each node, and kubelet already created mirror pod and report to API server.

**Comment by gmarek** [comment]
@dchen1107 - we should and at some point I tried to make reboot tests work this way, but I obviously failed (or maybe I did it only for some subset of tests - I'm not quite sure).

**Comment by dchen1107** [comment]
@gmarek I just saw the latest failure. let me take over this one.

**Comment by gmarek** [comment]
It's still happening. #20686 didn't help with it, as the problem is that system pods started from manifest are not listed in the API server. The logic added in #20686 was already there, though for all pods, not only kube-system ones: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/scheduler_predicates.go#L157.

We need to do something harder if we want to keep this test (and I think we should). Can Kubelet report itself as NotReady until it starts all system Pods? @dchen1107

**Comment by gmarek** [comment]
I also think that we should revert #20686, as it duplicates existing functionality.

**Comment by dchen1107** [comment]
How can kubelet knows which kube-system pods belonging to which node? Not all kube-system pods are static pods, especially we want to get rid of all left static kube-system pods. 

My problem is actually with the test written like this. I talked to @davidopp offline. we should rewrite this test properly by creating a deterministic environment since it requires one. I re-assign this back to @davidopp

**Comment by gmarek** [comment]
@dchen1107 - do you have an idea how to 'create a deterministic environment' in a running cluster in other way than running those tests sequentially and waiting for all non-system namespaces to be deleted?

We can create completely artificial environment and rewrite this into integration tests, but I don't think it's a good idea. The situation in which we currently are, i.e. we can't make cluster stable for few minutes, because something fails on one of three nodes is bad. This test failing is only a symptom of this.

**Comment by davidopp** [comment]
FWIW I think @dchen1107 actually was suggesting that we rewrite this one as well as #22181 and #22189 as integration tests. It might be OK to do it but I haven't had a chance to look at the tests carefully yet.

**Comment by gmarek** [comment]
Rewriting this into integration tests will certainly help. The question is are we happy with the state, when we can't make the cluster stable.

**Comment by yujuhong** [comment]
> The situation in which we currently are, i.e. we can't make cluster stable for few minutes, because something fails on one of three nodes is bad. This test failing is only a symptom of this.

How often does this test fail after adding the "waiting for the cluster to stabilize" function and why does it fail? I think for the majority of the time, this test should pass in our test environment, assuming nothing else can assign pods to nodes.

**Comment by gmarek** [comment]
We don't have this data yet - waiting for namespace deletion was added two days ago.

**Comment by yujuhong** [comment]
Let's keep this on hold until we have more data...

**Comment by gmarek** [comment]
It failed in gke suite. It seems to be some other kind of error. Wrote #22440 to help investigate.

**Comment by gmarek** [comment]
I haven't seen this failure for quite a while.
